File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        int[] sum = new int [n+1];
        if (n == 0) {
            return 0;
        } else if (n == 1) {
            return 1;
        }
        sum[0] = 0;
        sum[1] = 1;
        for (int i = 2; i < n+1; i++) {
            sum[i] = sum[i-1] + sum[i-2];
        }
        return sum[n];
    }
}

File: 0509-fibonacci-number/0509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        vector<int> nums;
        
        nums.push_back(0);
        nums.push_back(1);
        
        for (int i = 2; i <= n; i++) {
            nums.push_back(nums.at(i-1) + nums.at(i-2));
        }
        
        return nums.at(n);
    }
};